src/main/java/dev/codesupport/web/common/service/validation/persistant/AbstractPersistenceValidation.java

```diff
@@ -20,15 +23,34 @@
  * @see IdentifiableEntity
  * @see IdentifiableDomain
  */
+@Data
 public abstract class AbstractPersistenceValidation<E extends IdentifiableEntity<I>, I, D extends IdentifiableDomain<I>, C extends CrudRepository<E, I>> {
 
+    private final Class<E> entityType;
+
     /**
      * The associated repository for the given resource
      */
     protected C repository;
 
     public AbstractPersistenceValidation(C repository) {
         this.repository = repository;
+
+        Object type = this.getClass().getGenericSuperclass();
+
+        // Gets the Entity type defined by the child class.
+        if (type instanceof ParameterizedType) {
+            Object clazz = ((ParameterizedType) type).getActualTypeArguments()[0];
```

Comment:
Is `clazz` the best name?

Reply:
can't use ``class``, because it's a special keyword, ``clazz`` is the convention i've seen many libraries use.